frontend/src/metabase/sharing/components/PulsesListSidebar.jsx

```diff
@@ -0,0 +1,214 @@
+              size={18}
+              onClick={createSubscription}
+            />
+          </Tooltip>
+          <Tooltip tooltip={t`Close`}>
+            <Icon
+              name="close"
+              className="text-light bg-light-hover rounded p1 cursor-pointer"
+              size={22}
+              onClick={onCancel}
+            />
+          </Tooltip>
+        </Flex>
+      </div>
+      <div className="my2 mx4">
+        {pulses.map(pulse => (
+          <Card
+            key={pulse.id}
+            flat
+            className="mb3 cursor-pointer bg-brand-hover"
+            onClick={() => editPulse(pulse, pulse.channels[0].channel_type)}
+          >
+            <div className="px3 py2 hover-parent hover--inherit text-white-hover">
+              <div className="flex align-center hover-child hover--inherit">
+                <Icon
+                  name={
+                    pulse.channels[0].channel_type === "email"
+                      ? "mail"
+                      : "slack"
+                  }
+                  className="mr1"
+                  style={{ paddingBottom: "5px" }}
+                  size={16}
+                />
+                <Label className="hover-child hover--inherit">
+                  {friendlySchedule(pulse.channels[0])}
+                </Label>
+              </div>
+              {pulse.channels[0].channel_type === "email" && (
+                <EmailRecipients pulse={pulse} />
+              )}
+            </div>
+          </Card>
+        ))}
+      </div>
+    </Sidebar>
+  );
+}
+
+PulsesListSidebar.propTypes = {
+  pulses: PropTypes.array.isRequired,
+  createSubscription: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
+  editPulse: PropTypes.func.isRequired,
+};
+
+function EmailRecipients({ pulse }) {
+  const recipients = pulse.channels[0].recipients;
+  const [first, ...rest] = recipients;
+
+  let text = "";
+
+  if (rest != null && rest.length > 0) {
+    text += ngettext(
+      msgid` and ${rest.length} other`,
+      ` and ${rest.length} others`,
+      rest.length,
+    );
+  }
+
+  return (
+    <div className="text-medium hover-child">
+      <ul
+        className="flex flex-wrap scroll-x scroll-y"
+        style={{ maxHeight: 130 }}
+      >
+        <li className="flex align-center mr1 text-bold text-medium hover-child hover--inherit">
+          <Icon
+            name="group"
+            className="text-medium hover-child hover--inherit"
+            size={12}
+          />
+          <span
+            className="ml1 text-medium hover-child hover--inherit"
+            style={{ fontSize: "12px" }}
+          >
+            {first.common_name || first.email}
+            {text !== "" && text}
+          </span>
+        </li>
+      </ul>
+    </div>
+  );
+}
+
+EmailRecipients.propTypes = {
+  pulse: PropTypes.object.isRequired,
+};
+
+function friendlySchedule(channel) {
```

Comment:
I didn't move `friendlySchedule` as it depends on `channel`, but I moved the functions it depends on to `metabase/lib/time` since they are free of any dependencies on this subscription logic.